torch/distributed/_tensor/ops/utils.py

```diff
@@ -69,3 +70,33 @@ def normalize_dims(dims: Union[int, Sequence[int]], ndim: int) -> Sequence[int]:
 
 def prod(xs: Iterable[int]) -> int:
     return functools.reduce(operator.mul, xs, 1)
+
+
+def is_tensor_shardable(shape: Sequence[int], spec: DTensorSpec) -> bool:
+    """
+    Check if the shape is shardable according to the spec.
+    """
+    # number of shards in each tensor dimension
+    shards_map = [1] * len(shape)
+    for i, placement in enumerate(spec.placements):
+        if placement.is_shard():
+            shard_dim = cast(Shard, placement).dim
+            shards_map[shard_dim] *= spec.mesh.size(i)
+
+    for i, dim_size in enumerate(shape):
+        # TODO: maybe we should determine is_sharable base on the
+        #      whether it's evenly sharded or not
```

Comment:
Is the indentation correct here? Can polish the `TODO` information:
"is_sharable" -> "is_shardable"
"base" -> "based"
"on the whether" -> "on whether"